sycl/include/sycl/ext/oneapi/experimental/group_load_store.hpp

```diff
@@ -0,0 +1,1006 @@
+                              InputIteratorT>::value_type>,
+                          OutputT> &&
+    detail::is_generic_group_v<Group>>
+group_load(Group g, InputIteratorT in_ptr, OutputT &out,
+           Properties properties = {}) {
+  group_load(g, in_ptr, span<OutputT, 1>(&out, 1), properties);
+}
+
+// Store API scalar.
+template <typename Group, typename InputT, typename OutputIteratorT,
+          typename Properties = decltype(properties())>
+std::enable_if_t<std::is_convertible_v<
+                     InputT, remove_decoration_t<typename std::iterator_traits<
+                                 OutputIteratorT>::value_type>> &&
+                 detail::is_generic_group_v<Group>>
+group_store(Group g, const InputT &in, OutputIteratorT out_ptr,
+            Properties properties = {}) {
+  group_store(g, span<const InputT, 1>(&in, 1), out_ptr, properties);
+}
+
+// Load API sycl::vec overload.
+template <typename Group, typename InputIteratorT, typename OutputT, int N,
+          typename Properties = decltype(properties())>
+std::enable_if_t<
+    std::is_convertible_v<remove_decoration_t<typename std::iterator_traits<
+                              InputIteratorT>::value_type>,
+                          OutputT> &&
+    detail::is_generic_group_v<Group>>
+group_load(Group g, InputIteratorT in_ptr, sycl::vec<OutputT, N> &out,
+           Properties properties = {}) {
+  group_load(g, in_ptr, span<OutputT, N>(&out[0], N), properties);
+}
+
+// Store API sycl::vec overload.
+template <typename Group, typename InputT, int N, typename OutputIteratorT,
+          typename Properties = decltype(properties())>
+std::enable_if_t<std::is_convertible_v<
+                     InputT, remove_decoration_t<typename std::iterator_traits<
+                                 OutputIteratorT>::value_type>> &&
+                 detail::is_generic_group_v<Group>>
+group_store(Group g, const sycl::vec<InputT, N> &in, OutputIteratorT out_ptr,
+            Properties properties = {}) {
+  group_store(g, span<const InputT, N>(&in[0], N), out_ptr, properties);
+}
+
+// Load API span + group/sub_group overload.
+template <typename GroupHelper, typename InputIteratorT, typename OutputT,
+          std::size_t ElementsPerWorkItem,
+          typename Properties = decltype(properties())>
+std::enable_if_t<
+    std::is_convertible_v<remove_decoration_t<typename std::iterator_traits<
+                              InputIteratorT>::value_type>,
+                          OutputT> &&
+    detail::is_generic_group_v<GroupHelper>>
+group_load(GroupHelper gh, InputIteratorT in_ptr,
+           span<OutputT, ElementsPerWorkItem> out, Properties properties = {}) {
+  constexpr bool blocked = detail::is_blocked(properties);
+
+  using value_type = remove_decoration_t<
+      typename std::iterator_traits<InputIteratorT>::value_type>;
+
+  // See std::enable_if_t above restricting this implementation.
+  using GroupTy = GroupHelper;
```

Comment:
I'm not sure why we can't just name it `Group` (or `GroupTy` or `GroupTy`) in the template. I would argue `GroupHelper` is even more confusing as it could also suggest that the type is some intermediate helper type.